torch_geometric/nn/conv/sage_conv.py

```diff
@@ -45,30 +54,45 @@ class SAGEConv(MessagePassing):
         - **outputs:** node features :math:`(|\mathcal{V}|, F_{out})` or
           :math:`(|\mathcal{V_t}|, F_{out})` if bipartite
     """
-    def __init__(self, in_channels: Union[int, Tuple[int, int]],
-                 out_channels: int, normalize: bool = False,
+    def __init__(self, in_channels: Union[int, Tuple[int,
+                                                     int]], out_channels: int,
+                 aggregator_type: str = 'mean', normalize: bool = False,
                  root_weight: bool = True, bias: bool = True, **kwargs):
-        kwargs.setdefault('aggr', 'mean')
         super().__init__(**kwargs)
 
         self.in_channels = in_channels
         self.out_channels = out_channels
         self.normalize = normalize
         self.root_weight = root_weight
+        self.aggregator_type = aggregator_type
+
+        assert self.aggregator_type in ['mean', 'max', 'lstm', 'gcn', None]
 
         if isinstance(in_channels, int):
             in_channels = (in_channels, in_channels)
 
-        self.lin_l = Linear(in_channels[0], out_channels, bias=bias)
+        if self.aggregator_type == 'gcn':
```

Comment:
I'm not sure why we need `gcn` style aggregation here. This is already implemented in `GCNConv`.

Reply:
I added that because the paper has "GraphSAGE-GCN" as well. But I can definitely remove that since the same behaviour can be achieved by using:
```SAGEConv(in_channels, out_channels, aggr='mean', root_weight=False)```